apple/internal/partials/framework_import.bzl

```diff
@@ -69,28 +82,52 @@ def _framework_import_partial_impl(ctx, targets, targets_to_avoid):
         if framework_relative_dir:
             parent_dir = paths.join(parent_dir, framework_relative_dir)
 
-        bundle_files.append(
-            (processor.location.framework, parent_dir, depset([file])),
-        )
+        if paths.replace_extension(parent_dir, "") == file.basename:
```

Comment:
I think I'm missing something, what is this test looking specifically for?

Reply:
I'll have to leave that to @sergiocampama  to answer sorry!

Reply:
Talking with @allevato we realized what this was after, it's checking to see if the the parent is "Foo.[extension]" and the file is "Foo", thus "Foo.framework/Foo", so the binary within the framework.

So this is correct for iOS/tvOS, but with the symlinks that happen for macOS frameworks, this would be 100% correct.  So please add a comment here for what it is doing, and that it might need a revisit if macOS ever got multiple arches and and desired this support or some other platform uses a different framework layout compared.